ext/cgi/escape/escape.c

```diff
@@ -376,22 +376,65 @@ accept_charset(int argc, VALUE *argv, VALUE self)
  *  call-seq:
  *     CGI.unescape(string, encoding=@@accept_charset) -> string
  *
- *  Returns URL-unescaped string.
+ *  Returns URL-unescaped string (+application/x-www-form-urlencoded+).
+ *
+ */
+static VALUE
+cgiesc_decode_www_form_component(int argc, VALUE *argv, VALUE self)
+{
+    VALUE str = (rb_check_arity(argc, 1, 2), argv[0]);
+
+    StringValue(str);
+
+    if (rb_enc_str_asciicompat_p(str)) {
+        VALUE enc = accept_charset(argc-1, argv+1, self);
+        return optimized_unescape(str, enc, 1);
+    }
+    else {
+        return rb_call_super(argc, argv);
+    }
+}
+
+/*
+ *  call-seq:
+ *     CGI.url_encode(string) -> string
+ *
+ *  Returns URL-escaped string  following RFC 3986.
```

Comment:
```suggestion
 *  Returns URL-escaped string following RFC 3986.
```

Minor: is that a duplicate space?